tests/mocha/workspace_svg_test.js

```diff
@@ -116,6 +116,171 @@ suite('WorkspaceSvg', function() {
+    });
+    teardown(function() {
+      delete Blockly.Blocks['stack_block'];
+    });
+
+    test('inject', function() {
+      // TODO: Address and remove before merging PR
+      // How do we want this to behave?
+      // Change listeners are attached after resizes, but events are disabled
+      // so viewport event is fired, but not listened to (at this time).
+      workspaceTeardown.call(this, this.workspace);
+      var workspace = Blockly.inject('blocklyDiv');
+      this.clock.runAll();
+      assertViewportEventFired(
+          this.eventsFireStub, this.changeListenerSpy, workspace);
+    });
+    test('dispose fires no viewport event', function() {
+      // TODO: Address and remove before merging PR
+      // Do we want this test?
+      this.eventsFireStub.resetHistory();
+      this.changeListenerSpy.resetHistory();
+      this.workspace.dispose();
+      assertEventNotFired(
+          this.changeListenerSpy, Blockly.Events.Ui, {element: 'viewport'});
+    });
+
+    suite('zoom', function() {
+      test('setScale', function() {
+        runViewportEventTest(() => this.workspace.setScale(2),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('zoom(50, 50, 1)', function() {
+        runViewportEventTest(() => this.workspace.zoom(50, 50, 1),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('zoom(50, 50, -1)', function() {
+        runViewportEventTest(() => this.workspace.zoom(50, 50, -1),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('zoomCenter(1)', function() {
+        runViewportEventTest(() => this.workspace.zoomCenter(1),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('zoomCenter(-1)', function() {
+        runViewportEventTest(() => this.workspace.zoomCenter(-1),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('zoomToFit', function() {
+        var block = this.workspace.newBlock('stack_block');
+        block.initSvg();
+        block.render();
+        runViewportEventTest(() => this.workspace.zoomToFit(),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+    });
+    suite('scroll', function() {
+      test('centerOnBlock', function() {
+        var block = this.workspace.newBlock('stack_block');
+        block.initSvg();
+        block.render();
+        runViewportEventTest(() => this.workspace.zoomToFit(block.id),
+            this.eventsFireStub, this.changeListenerSpy, this.workspace,
+            this.clock);
+      });
+      test('scroll', function() {
+        runViewportEventTest(() => this.workspace.scrollCenter(),
```

Comment:
This test and the next are the same.

Reply:
That's a mistake, I'll update it to call scroll as intended.